osquery/tables/applications/vscode_extensions.cpp

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <set>
+#include <string>
+
+#include <boost/filesystem.hpp>
+
+#include <osquery/core/tables.h>
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/logger/logger.h>
+#include <osquery/tables/system/system_utils.h>
+#include <osquery/utils/json/json.h>
+
+namespace fs = boost::filesystem;
+
+namespace osquery {
+namespace tables {
+
+namespace {
+std::string getStringValuefromRjObject(const rapidjson::Value& obj,
+                                       const std::string& key) {
+  rapidjson::Value::ConstMemberIterator itr = obj.FindMember(key);
+  if (itr != obj.MemberEnd()) {
+    const rapidjson::Value& value = itr->value;
+
+    if (value.IsString()) {
+      return value.GetString();
+    }
+
+    if (value.IsInt64()) {
+      return std::to_string(value.GetInt64());
+    }
+
+    if (value.IsBool()) {
+      return value.GetBool() ? "1" : "0";
+    }
+  }
+
+  return "";
+}
+} // namespace
+
+void genReadJSONAndAddExtensionRows(const std::string& uid,
+                                    const std::string& path,
+                                    QueryData& results) {
+  std::string json;
+  if (!readFile(path, json).ok()) {
+    LOG(WARNING) << "Could not read vscode extensions.json from '" << path
```

Comment:
I think this should be logged at debug level as it will generate a ton of warnings if you try to query every user on a device (eg. `select * from users cross join  vscode_extensions using (uid);`)